### PR TITLE
Added github actions

### PR DESCRIPTION
**Pull request by lzap**
SSIA

Replaces https://github.com/logicminds/rubyipmi/pull/52

I think it won't run until this is merged. I am commited to fixing this if it does not end up green, I haven't figured out how to run these without merging into `master`.

### CODE REVIEW

**Comment by lzap** [comment]
Would you mind adding our project email on rubygems.org?

https://rubygems.org/profiles/theforeman

You can keep handling the releases but just for case. Our community has strict rules: e.g. 2FA on both github and rubygems, devs need to be respected and nominated in order to have push access and overall we are a great community! :-)

File: .github/workflows/test.yml.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      fail-fast: false
+      matrix:
+        include:
+          - ruby: "2.5"
+          - ruby: "2.6"
+          - ruby: "2.7"
+          - ruby: "3.0"
+            coverage: "yes"
+    env:
+      COVERAGE: ${{ matrix.coverage }}
```
Comment (ekohl) [inline .github/workflows/test.yml.erb:20]: This was set up specifically in combination with `Gemfile` and rake tasks. Here that isn't present so you can simplify it.
```suggestion
        ruby:
          - "2.5"
          - "2.6"
          - "2.7"
          - "3.0"
```

**Review by ekohl — COMMENTED** [review]
If you're only testing on 2.5+ I'd suggest adding a minimum gem version in gemspec via `s.required_ruby_version = '>= 2.5'`.

There's also a trick to get GHA to run on the first PR: you can push the same commit to a branch in this repo. That starts CI. After that you can immediately delete the branch again. It will also continue to run CI even on new pushes.

**Comment by logicminds** [comment]
> https://rubygems.org/profiles/theforeman

@lzap Where would this go?

**Pull request merged**